api/deepcompare/equal.go

```diff
@@ -0,0 +1,59 @@
+package deepcompare
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+
+	"github.com/gogo/protobuf/types"
+)
+
+// Comparable can be implemented if an object knows how to compare another into itself.
+type Comparable interface {
+	Equal(other interface{}) bool
+}
+
+// Equal does a deep compare between msg1 and msg2.
+//
+// If the type has an Equal function defined, it will be used.
+//
+// Default implementations for builtin types and well known protobuf types may
+// be provided.
+//
+// If the comparison cannot be performed, this function will panic. Make sure
+// to test types that use this function.
+func Equal(msg1, msg2 interface{}) bool {
+	switch msg1 := msg1.(type) {
+	case *types.Any:
```

Comment:
Non-blocking:  it seems like the `*types.X` types implement the `Comparable` interface, at least in our vendored code.  Do we still need to provide these implementations?